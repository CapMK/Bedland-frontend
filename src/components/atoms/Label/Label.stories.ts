import type { Meta, StoryObj } from '@storybook/react';
import Label from './Label';

const meta = {
  title: 'Atoms/Label',
  component: Label,
  tags: ['autodocs'],
} satisfies Meta<typeof Label>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    label: 'login',
    htmlFor: 'login',
    textColor: 'white',
    size: 'small',
  },
};
