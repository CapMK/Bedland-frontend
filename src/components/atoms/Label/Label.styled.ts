import styled from 'styled-components';

const sizes: { [key: string]: number } = { small: 10, medium: 16, large: 24 };

export const Lbl = styled.label<{
  textColor: string;
  size: 'small' | 'medium' | 'large';
}>`
  color: ${(props) => props.textColor};
  display: flex;
  font-size: ${(props) => sizes[props.size]};
`;

export default Lbl;
