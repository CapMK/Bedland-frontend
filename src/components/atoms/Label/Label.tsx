import * as Styled from './Label.styled';

type LabelProps = {
  htmlFor: string;
  label: string;
  textColor: string;
  size: 'small' | 'medium' | 'large';
};

const Label = ({ htmlFor, label, textColor, size }: LabelProps) => (
  <Styled.Lbl htmlFor={htmlFor} textColor={textColor} size={size}>
    {label}
  </Styled.Lbl>
);

export default Label;
