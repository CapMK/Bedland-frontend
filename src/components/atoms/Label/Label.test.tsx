import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Label from './Label';

describe('Label', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderer
      .create(<Label label='login' htmlFor='login' textColor='white' size='small' />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
