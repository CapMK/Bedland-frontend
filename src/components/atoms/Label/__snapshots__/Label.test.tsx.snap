// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > renders correctly 1`] = `
.c0 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 10;
}

<label
  className="c0"
  htmlFor="login"
  size="small"
>
  login
</label>
`;
